src/services/api/dock-api.service.ts

```diff
@@ -0,0 +1,105 @@
+import * as dotenv from 'dotenv';
+import { ProviderService } from './provider.service.js';
+
+dotenv.config();
+
```

Comment:
All the providers should have to implement a common interface similar to our credo-ts, acapy work.

```
export interface IProviderService {
public readonly supportedProvider = 'dock';
  public static instance = new NewProviderApiService('NEW_PROVIDER_MASTER_API_KEY');

  protected getDefaultApiUrl(): string {
    return 'https://api.newprovider.com';
  }

```